_episodes/08-development-setup.md

```diff
@@ -1,138 +1,461 @@
+install the development dependencies. We should check if the installation
+works properly. To do this, run the tool with:
+
+~~~bash
+esmvaltool --help
+~~~
+
+If installation is successful, ESMValTool prints a help message to the console.
+For more details about development installation, see ESMValTool documentation on
+[install from source][install-from-source].
+
+> ## Checking the development installation
+>
+> We can use the command ``conda list`` to list installed packages in
+> the ``esmvaltool`` environment.
+> Use this command to check that ESMValTool is installed in a ``develop`` mode.
+>
+> **Tips**: see the
+> [documentation on conda list](https://docs.conda.io/projects/conda/en/latest/commands/list.html).
+>
+>> ## Solution
+>>
+>> Run:
+>>
+>> ~~~bash
+>> conda list esmvaltool
+>>~~~
+>>
+>> ~~~bash
+>> # Name                    Version                   Build  Channel
+>> esmvaltool                2.1.1                     dev_0    <develop>
+>>~~~
+>>
+>> As can be seen, it is mentioned ``<develop>`` under the ``Channel``.
+> {: .solution}
+{: .challenge}
+
+## Contribution
+
+We have seen how to install ESMValTool in a ``develop`` mode.
+Now, we try to contribute to its development. Let's see how we can achieve this.
+
+### Review process
+
+We first discuss our ideas in an
+**[issue](https://github.com/ESMValGroup/ESMValTool/issues)** in ESMValTool repository.
+This can avoid disappointment at a later stage, for example,
+if more people are doing the same thing.
+It also gives other people an early opportunity to provide input and suggestions,
+which results in more valuable contributions.
+
+Then, we create a new ``branch`` locally and start developing new codes.
+Once our development is finished, we can initiate a ``pull request``.
+For a full description of the GitHub workflow, please see ESMValTool documentation on
+[GitHub Workflow](https://docs.esmvaltool.org/en/latest/community/repository.html#github-workflow).
+
+The pull request will be tested, discussed and merged. This is called "**review process**".
+The process will take some effort and time to learn.
+However, a few “tools” i.e. command lines gets you a long way,
+and we’ll cover those essentials in the next sections.
+
+**Tips**: we encourage you to keep the pull requests small.
+Reviewing small incremental changes are more efficient.
+
+### Background
+
+We saw 'warming stripes' in lesson
+[Writing your own recipe]({{ page.root }}{% link _episodes/05-preprocessor.md %}).
+Imagine the following task: you want to contribute warming stripes recipe and diagnostics
+to ESMValTool. You have to add the diagnostics
+[warming_stripes.py](../files/warming_stripes.py) and the recipe
+[recipe_warming_stripes.yml](../files/recipe_warming_stripes.yml)
+to their locations in ESMValTool directory.
+After these changes, you should also check if everthing works fine.
+This is where we take advantage of the tools that are introduced later.
```

Comment:
Just a development tip from my side: I'm using the 'rewrap' extension in vscode. This let's you auto-wrap lines with `alt+Q` so you don't exceed the line length. I find that super useful for these kind of text blocks.

Reply:
I just tried this extension, very cool, thank you for the suggestion.